### DISCUSSION TITLE
Is there benchmark for hyperscript version of solid?

### DISCUSSION THREAD

**Discussion by zhaoyao91** [solidjs/solid#455] [Q&A] [answered]
no-compile is very appelling for our project. but mentioned in the readme, hyperscript version of solid is not as performant as jsx version of it. I am curious about how much the difference is to make some critical project decisions. Wish there be some result like this [benchmark](https://github.com/krausest/js-framework-benchmark)

**Comment by ryansolid** [comment, marked as answer]
Read this: https://betterprogramming.pub/the-fastest-way-to-render-the-dom-e3b226b15ca3. It's outdated but should be similar still. All non-signal implementations use state so they could be 0.04 faster in theory. Solid's a little smaller now then back then too. So might shave 1kb across the board.

**Reply by zhaoyao91** [reply]
Oops, seems there are notable gaps in performance between jsx version and hyper script version. Thanks for this great comparison.

**Reply by ryansolid** [reply]
I mean it's still decently good performance. Like Svelte range. But yes it is noticeably slower since we don't get to leverage compiler analysis. That shouldn't be that big of a detractor. I actually think the bigger strike is the developer experience. Having to manually manage wrapping expressions is easy to mess up etc...

**Reply by zhaoyao91** [reply]
Yeah, I found it, when trying to make some benchmark for solid-h, I felt it is not fluent to write hyper components.
solid-html may be a good compromise, but the string literal is always compressable, which again requires us to add compile time tools ...

**Reply by ryansolid** [reply]
Although I did solid-html as a runtime approach. It is much more optimal than hyperscript because it can do structural analysis at runtime (although not all available to JSX). But it still does all the manual wrapping. There is currently no compiler to give optimal output, you'd be falling back to HyperScript more than likely. So even if you went solid-html you'd be stuck. I didn't go compile time with it since it would need to be different in a way that would never work for runtime and we already have JSX for that. While compile time could help a bit on size and speed up initialization it wasn't something I considered worth the effort. Honestly I think solid-html is still missing a pretty important feature.. spreads which I haven't updated my runtime parser to handle. I think this might be a non-starter for a lot of things.
